drivers/gpio/gpio_ite_it8801.c

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright (c) 2024 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8801_gpio
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mfd/mfd_ite_it8801.h>
+#include <zephyr/spinlock.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_ite_it8801, CONFIG_GPIO_LOG_LEVEL);
+
+struct gpio_it8801_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* IT8801 controller dev */
+	const struct device *mfd;
+	/* I2C device for the MFD parent */
+	const struct i2c_dt_spec i2c_dev;
+	/* GPIO input pin status register */
+	uint8_t reg_ipsr;
+	/* GPIO set output value register */
+	uint8_t reg_sovr;
+	/* GPIO control register */
+	uint8_t reg_gpcr;
+	/* GPIO interrupt status register */
+	uint8_t reg_gpisr;
+	/* GPIO interrupt enable register */
+	uint8_t reg_gpier;
+	uint8_t pin_mask;
+};
+
+struct gpio_it8801_data {
+	struct gpio_driver_data common;
+	const struct device *gpio_dev;
+	sys_slist_t callbacks;
+	uint8_t level_high_trig;
+	uint8_t level_low_trig;
+};
+
+static int ioex_check_is_not_valid(const struct device *dev, gpio_pin_t pin)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret = 0;
+
+	if (BIT(pin) & ~(config->pin_mask)) {
+		LOG_ERR("GPIO  port%d-%d is not support", config->reg_ipsr, pin);
+		return -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+	uint8_t reg_gpcr = config->reg_gpcr + pin;
+	uint8_t mask = BIT(pin);
+	uint8_t new_value = 0, control;
```

Comment:
I think it'd be cleaner if you were to not initialize this one here and instead add a `else` case below that explicitly sets it to 0, then all possible assignment would be near each other

Reply:
Done.